WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt

```diff
@@ -131,8 +130,8 @@ class ReaderDiscoverLogic(
         val fullCardsJson = json.optJSONArray(JSON_CARDS)
 
         // Parse the json into cards model objects
-        val cards = parseCards(fullCardsJson)
-        insertPostsIntoDb(cards.filterIsInstance<ReaderPostCard>().map { it.post })
+        val postCards = parsePostCards(fullCardsJson)
+        insertPostsIntoDb(postCards.map { it.post })
```

Comment:
I've decided to rename the function and modify it to return only `ReaderPostCard` objects. 
It did confuse me before and other types are not used at all here so the code was unnecessary.